eclipsesource/theia
# Add initial support for variable resolution from user input

#### What it does

- Ensure we always create a variable part even for undefined variables
-- Prompt text will then default to user text (including '#')

- Allow adopters to register variable resolution contributions
-- Given a particular variable name, argument and location

- Automatically resolve all variable parts in a chat request
-- Ensure parts always provide a matching prompt text

Relates to https://github.com/eclipsesource/osweek-2024/issues/46

#### How to test

- Test that normal chat still works as before
- Use variable '#today' in your prompt and try different arguments such as 'inUnixSeconds' or 'inIso8601' combined as '#today:inIso8601' to see different results.

![ai-variables](https://github.com/user-attachments/assets/fe446e2f-22dc-4931-bb21-2a89b3dcc883)

#### Follow-ups



#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
